### ISSUE TITLE
Some Question About Experiment

### ISSUE THREAD

**Issue by zqp1226358** [CityU-AIM-Group/SFDA-FSM#7]
Hello, thank you for your reply.
Currently I run the code but the result is not what I expected. 

![image](https://github.com/CityU-AIM-Group/SFDA-FSM/assets/47075652/e6b71327-802d-46e3-8806-00da2e1bf856)

I suspect that there is still a problem with my preprocessing？
the coarse_generation is as follows
![image](https://github.com/CityU-AIM-Group/SFDA-FSM/assets/47075652/80a2dd1c-2f9a-4a1a-9b07-728d71a21d5e)

the fine_generation is as follows
![image](https://github.com/CityU-AIM-Group/SFDA-FSM/assets/47075652/61059a71-cc06-43d0-adfe-ca0fdc717152)

the pseudo label is as follows,  I pass in the source-like image to the pretrained model you provided
![image](https://github.com/CityU-AIM-Group/SFDA-FSM/assets/47075652/06ae2840-e8ae-41fe-8ab2-4b40ad25c39e)

In addition, The "resnet_pretrain.pth" is missing, Is it convenient for you to provide it?

Looking forward to your reply. Thank you very much.

**Comment by puyln** [comment]
Hi, sorry to bother you, I want to ask you a question about "inversed" data and "pseudo" data. What images are contained in these folders?

**Comment by zqp1226358** [comment]
Hi，@puyln  "inversed" data is fine_generation，"pseudo" data is generated by source-like image through pretained source model。

**Comment by qwerasdzxcvb** [comment]
Hello, I am reproducing your code and I am facing issues in Generation Stage：
![coarse](https://github.com/CityU-AIM-Group/SFDA-FSM/assets/110284993/a305ad33-8193-4024-acbb-2b66ccbeeff9)
Every coarse generate doesn't show any content.
![fine](https://github.com/CityU-AIM-Group/SFDA-FSM/assets/110284993/d49f23bd-4027-4129-85a4-6dad11ab5dec)
 I could not find a way to solve this issue, can you please provide some help？
Thanks for your time.

**Comment by zqp1226358** [comment]
@qwerasdzxcvb 
you can look this, in code try to initialize the input noise with image content.
https://github.com/CityU-AIM-Group/SFDA-FSM/issues/2#issuecomment-1352951661

**Comment by qwerasdzxcvb** [comment]
Thank you for your reply, but the issue still persists after changing to content image.
![coarse1](https://github.com/CityU-AIM-Group/SFDA-FSM/assets/110284993/c522195a-cd20-431a-aa88-a3c59f53f6db)
Is it convenient to leave a contact information? I have some specific questions I'd like to ask you.

**Comment by qwerasdzxcvb** [comment]
@zqp1226358 
Thank you for your reply, but the issue still persists after changing to content image.
![coarse1](https://github.com/CityU-AIM-Group/SFDA-FSM/assets/110284993/8459e509-e680-40ef-8fdc-79c2e81ba4e8)
Is it convenient to leave a contact information? I have some specific questions I'd like to ask you.

**Comment by zqp1226358** [comment]
@qwerasdzxcvb You can contact me by [google email](zhuqpe@gmail.com)

**Comment by Mickey-Boy** [comment]
@zqp1226358 Thanks for your answers. @qwerasdzxcvb You'd better initialize the input noise image by original image as suggested in #2 .

**Comment by cyang-cityu** [comment]
@zqp1226358 Thanks for your answers. @qwerasdzxcvb You'd better initialize the input noise image with original image as suggested by #2.

**Comment by cyang-cityu** [comment]
@zqp1226358 This is the [resnet pretrained model](https://drive.google.com/file/d/1LS_5gRc9B0-BXf_OrZfVbidn0JmNHrws/view?usp=sharing). You can also apply our pretrained source model for coarse generation.

**Comment by WenTaoLiu502** [comment]
hello，@zqp1226358 
I'm encountering the same problem during training. When I printed out the individual sub-losses, I found that the loss_pseudo one was significantly larger than the remaining two (i.e., loss_distill, loss_contrast), perhaps a potential cause of the gradient explosion. When I turned off the compactness metric, this phenomenon disappeared, but the Dice score still lags behind the result in paper by a lot (i.e., 45.703 vs 71.074) . I also noticed that the learning rate in the source code was different from the paper (i.e., 1e-3 vs 5e-4), and when I fixed this configuration, unfortunately it still didn't work.
![2024-11-19 09-59-30屏幕截图](https://github.com/user-attachments/assets/f59ffe97-6348-40c0-bff7-0bf72546decc)
